src/components/Form.jsx

```diff
@@ -237,6 +238,23 @@ export default class Form extends React.Component {
     }));
   }
 
+  /**
+   * Handles callback. Conditionaly return props with Immutable
+   */
+  handleCallback = (callback, props) => {
```

Comment:
Please note that there are multiple places where immutable instances are converted to mutable (i.e. in various `fieldUtils`). I think we should move this function wrapper into a util function itself, to be able to wrap anything.

What do you think about this:
```js
// utils/withImmutable.js
import { Iterable } from 'immutable';

export default function withImmutable(func, params, context) {
  const { withImmutable } = context;

  /* When Immutable params allowed, bypass any transformation */
  resolvedParams = withImmutable ? params : params.reduce((nextParams, paramName) => {
    const paramValue = params[paramName];
    nextParams[paramName] = Iterable.isIterable(paramValue) ? paramValue.toJS() : paramValue;
    return nextParams;
  }, {});

  return func(resolvedParams);
}
```

Note that `Iterable.isIterable()` is a more generic way to determine the instances of Immutable. This will return `true` for `Map`, `List` and other Immutable instances.